packages/hash/frontend/src/pages/[account-slug]/entity-types/property-menu.tsx

```diff
@@ -0,0 +1,143 @@
+import { faEllipsis } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@hashintel/hash-design-system/fontawesome-icon";
+import {
+  IconButton,
+  IconButtonProps,
+} from "@hashintel/hash-design-system/icon-button";
+import { Menu } from "@hashintel/hash-design-system/menu";
+import { MenuItem } from "@hashintel/hash-design-system/menu-item";
+import {
+  Divider,
+  ListItem,
+  listItemClasses,
+  ListItemText,
+  listItemTextClasses,
+  menuItemClasses,
+  tableRowClasses,
+  Typography,
+} from "@mui/material";
+import {
+  bindMenu,
+  bindTrigger,
+  usePopupState,
+} from "material-ui-popup-state/hooks";
+import { useId } from "react";
+import { OntologyChip } from "./ontology-chip";
+import { PlaceholderIcon } from "./placeholder-icon";
+
+export const PropertyMenu = ({
+  disabled,
+  onRemove,
+  ...props
+}: IconButtonProps & { onRemove?: () => void }) => {
+  const id = useId();
+  const popupState = usePopupState({
+    variant: "popover",
+    popupId: `property-${id}`,
+  });
+
+  return (
+    <>
+      <IconButton
+        {...props}
+        disabled={disabled}
+        sx={{
+          opacity: 0,
+          ...(!disabled && {
+            [`.${tableRowClasses.root}:hover &`]: {
+              opacity: 1,
+            },
+          }),
+        }}
```

Comment:
I think a transition here would polish this a little bit, what do you think?

```suggestion
        sx={({ transitions }) => ({
          opacity: 0,
          transition: transitions.create("opacity"),
          ...(!disabled && {
            [`.${tableRowClasses.root}:hover &`]: {
              opacity: 1,
            },
          }),
        })}
```